modules/security/src/main/java/io/oasp/module/security/common/api/accesscontrol/AccessControlGroup.java

```diff
@@ -141,20 +135,11 @@ public boolean equals(Object obj) {
     if (!super.equals(obj)) {
       return false;
     }
-    if (getClass() != obj.getClass()) {
-      return false;
-    }
     AccessControlGroup other = (AccessControlGroup) obj;
```

Comment:
The `super.equals(...)` method will perform an additional `this == obj` check, but it will also check for `null` and for `class` equality - and in addition the super class' properties (here `id`) will be checked. Therefore, this shortens the code of `AccessControlGroup.equals(...)`.

Regarding the `ClassCastException` a short test method will show that there is no problem:
```java
  @Test
  public void testSuperClassType() {

    // setup
    AccessControlGroup acg = new AccessControlGroup();
    AccessControl ac = new MyAccessControl();

    // verify
    assertThat(acg.equals(ac)).isEqualTo(false);
    assertThat(ac.equals(acg)).isEqualTo(false);
  }
```

From my point of view this should be considered correct.